aspnetcore/security/authentication/windowsauth.md

```diff
@@ -82,43 +71,34 @@ Disable Anonymous Authentication and enable Windows Authentication.
 
 ![Visual Studio Publish Dialog](windowsauth/_static/vs-publish-app.png)
 
 Learn more about [publishing to IIS](xref:publishing/iis).
 
 Launch the app to verify Windows authentication is working.
 
-## Enabling Windows authentication with WebListener
+## Enable Windows authentication with HTTP.sys or WebListener
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+Although Kestrel doesn't support Windows authentication, you can use [HTTP.sys](xref:fundamentals/servers/httpsys) to support self-hosted scenarios on Windows. The following example configures the app's web host to use HTTP.sys with Windows authentication:
+
+[!code-csharp[](windowsauth/sample/Program2x.cs?highlight=9-14)]
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 Although Kestrel doesn't support Windows authentication, you can use [WebListener](xref:fundamentals/servers/weblistener) to support self-hosted scenarios on Windows. The following example configures the app's web host to use WebListener with Windows authentication:
 
-```csharp
-public class Program
-{
-    public static void Main(string[] args)
-    {
-        var host = new WebHostBuilder()
-            .UseWebListener(options =>
-            {
-                options.ListenerSettings.Authentication.Schemes = 
-                    AuthenticationSchemes.Negotiate | AuthenticationSchemes.NTLM;
-                options.ListenerSettings.Authentication.AllowAnonymous = false;
-            })
-            .UseContentRoot(Directory.GetCurrentDirectory())
-            .UseStartup<Startup>()
-            .Build();
-
-        host.Run();
-    }
-}
-```
-
-## Working with Windows authentication
-
-If your app uses Windows authentication and anonymous access, you can use the ``[Authorize]`` and ``[AllowAnonymous]`` attributes. Apps that do not have anonymous enabled do not require ``[Authorize]``; the  app is treated as requiring authentication, anonymous requests are rejected. Note, if the IIS site is configured **not** to allow anonymous access, the ``[AllowAnonymous]`` attribute does **not** allow anonymous requests. The ``[AllowAnonymous]`` attribute overrides ``[Authorize]`` attribute usage within apps that allow anonymous access.
+[!code-csharp[](windowsauth/sample/Program1x.cs?highlight=6-11)]
+
+---
+
+## Work with Windows authentication
+
+If your app allows Windows authentication and anonymous access, you can use the `[Authorize]` and `[AllowAnonymous]` attributes. Apps disallowing anonymous access don't require `[Authorize]`. In this scenario, the app is treated as requiring authentication, and anonymous requests are rejected. If the IIS site is configured to disallow anonymous access, the `[AllowAnonymous]` attribute doesn't allow anonymous requests. The `[AllowAnonymous]` attribute overrides `[Authorize]` attribute usage within apps which allow anonymous access. See [Simple Authorization](xref:security/authorization/simple) for more details.
```

Comment:
If the IIS site (or Http.Sys or WebListener servers) are configured to disallow anonymous access then the request never reaches your application and the `[AllowAnonymous]` attribute doesn't apply.

Reply:
Can we separate this into two sections? The simple case is that the servers disallow anonymous and no attributes are needed or applicable. If the servers support auth and anonymous then you need the attributes and one additional piece of config:
`services.AddAuthentication(IISDefaults.AuthenticationScheme);` or HttpSysDefaults.AuthenticationScheme though they have the same value. This only applies to 2.0. `[Authorize]` needs this to challenge anonymous requests for windows auth.